src/rpc/eth.rs

```diff
@@ -279,6 +282,8 @@ where
             let mut tracer = NoopTracer;
 
             let beneficiary = engine_factory(None, chain_spec, None)?.get_beneficiary(&header);
+            // TODO: check if we can reuse EvmMemory
+            let mut mem = EvmMemory::new();
```

Comment:
Here and in several other places `EvmMemory` is created inside method instead of reusing already existing instance. It can be expensive if such methods are called too often.